doc/en/how-to/assert.rst

```diff
@@ -282,6 +282,10 @@ exception at a specific level; exceptions contained directly in the top
 Alternate `pytest.raises` form (legacy)
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+.. warning::
```

Comment:
As discussed, let's remove this warning.

Removing the "Nonetheless" seems fine.

I also think we can remove the "The reporter will provide ..." paragraph to make the section shorter.